src/Stripe.net/Infrastructure/Public/StripeResponse.cs

```diff
@@ -1,13 +1,90 @@
 namespace Stripe
 {
     using System;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http.Headers;
 
+    /// <summary>
+    /// Represents a response from Stripe's API.
+    /// </summary>
     public class StripeResponse
     {
-        public string ResponseJson { get; set; }
+        /// <summary>Initializes a new instance of the <see cref="StripeResponse"/> class.</summary>
+        /// <param name="statusCode">The HTTP status code.</param>
+        /// <param name="headers">The HTTP headers of the response.</param>
+        /// <param name="content">The body of the response.</param>
+        public StripeResponse(HttpStatusCode statusCode, HttpResponseHeaders headers, string content)
+        {
+            this.StatusCode = statusCode;
+            this.Headers = headers;
+            this.Content = content;
+        }
 
-        public string RequestId { get; set; }
+        /// <summary>Gets the HTTP status code of the response.</summary>
+        /// <value>The HTTP status code of the response.</value>
+        public HttpStatusCode StatusCode { get; }
 
-        public DateTime RequestDate { get; set; }
+        /// <summary>Gets the HTTP headers of the response.</summary>
+        /// <value>The HTTP headers of the response.</value>
+        public HttpResponseHeaders Headers { get; }
+
+        /// <summary>Gets the body of the response.</summary>
+        /// <value>The body of the response.</value>
+        public string Content { get; }
+
+        /// <summary>Gets the date of the request, as returned by Stripe.</summary>
+        /// <value>The date of the request, as returned by Stripe.</value>
+        public DateTimeOffset? Date => this.Headers?.Date;
+
+        /// <summary>Gets the idempotency key of the request, as returned by Stripe.</summary>
+        /// <value>The idempotency key of the request, as returned by Stripe.</value>
+        public string IdempotencyKey => MaybeGetHeader(this.Headers, "Idempotency-Key");
+
+        /// <summary>Gets the ID of the request, as returned by Stripe.</summary>
+        /// <value>The ID of the request, as returned by Stripe.</value>
+        public string RequestId => MaybeGetHeader(this.Headers, "Request-Id");
+
+        /// <summary>
+        /// Gets the body of the response.
+        /// This method is deprecated and will be removed in a future version, please use the
+        /// <see cref="Content"/> property getter instead.
+        /// </summary>
+        /// <value>The body of the response.</value>
+        // TODO: remove this in a future a major version
+        [Obsolete("Use Content instead")]
+        public string ResponseJson => this.Content;
```

Comment:
Don't we use this for EphemeralKeys today? https://github.com/stripe/stripe-dotnet/blob/master/src/Stripe.net/Entities/EphemeralKeys/EphemeralKey.cs#L25 
We likely need to fix this too

Reply:
Yep, I did fix `EphemeralKey` :) Our StyleCop rules prevent us from calling obsolete methods in the library itself.